packages/mui-base/src/Unstable_NumberInput/NumberInput.test.tsx

```diff
@@ -498,32 +498,64 @@ describe('<NumberInput />', () => {
       expect(input.value).to.equal('1');
     });
 
-    it('sets value to min when the input has no value and ArrowUp is pressed', async () => {
-      const handleChange = spy();
+    describe('when the input has no value and ArrowUp is pressed', () => {
+      it('sets value to min if min is provided', async () => {
+        const handleChange = spy();
 
-      const { getByRole } = render(<NumberInput min={5} onChange={handleChange} />);
+        const { getByRole } = render(<NumberInput min={5} onChange={handleChange} />);
 
-      const input = getByRole('textbox') as HTMLInputElement;
+        const input = getByRole('textbox') as HTMLInputElement;
 
-      await userEvent.click(input);
-      await userEvent.keyboard('[ArrowUp]');
+        await userEvent.click(input);
+        await userEvent.keyboard('[ArrowUp]');
+
+        expect(handleChange.args[0][1]).to.equal(5);
+        expect(input.value).to.equal('5');
+      });
+
+      it('sets value to 1 if min is not provided', async () => {
+        const handleChange = spy();
+
+        const { getByRole } = render(<NumberInput onChange={handleChange} />);
 
-      expect(handleChange.args[0][1]).to.equal(5);
-      expect(input.value).to.equal('5');
+        const input = getByRole('textbox') as HTMLInputElement;
+
+        await userEvent.click(input);
+        await userEvent.keyboard('[ArrowUp]');
+
+        expect(handleChange.args[0][1]).to.equal(1);
+        expect(input.value).to.equal('1');
+      });
     });
 
-    it('sets value to max when the input has no value and ArrowDown is pressed', async () => {
-      const handleChange = spy();
+    describe('when the input has no value and ArrowDown is pressed', () => {
+      it('sets value to max when max is provided', async () => {
```

Comment:
This doesn't match the native behavior either - the `min` or `-1` value should be used here (see the first demo in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number)